tools/ci_build/github/android/build_aar_package.py

```diff
@@ -72,6 +72,38 @@ def _parse_build_settings(args):
     return build_settings
 
 
+# Add ORT C and C++ API headers to the AAR package (in fact a zip file)
+# Such that developers using ORT native API can extract libraries and header from AAR package without building ORT
+# TODO, see if we can use Gradle to add headers to AAR package directly, which is necessary if we want to
+# publish the packagee directly using Gradle in the pipeline
+def _add_headers_to_aar(aar_file_path):
+    import shutil
+    import tempfile
+    with tempfile.TemporaryDirectory() as temp_dir:
+        aar_content = os.path.join(temp_dir, 'aar_content')
+        shutil.unpack_archive(aar_file_path, aar_content, 'zip')
+        dst_headers_path = os.path.join(aar_content, 'headers')
+        os.makedirs(dst_headers_path)
+        src_headers_base_path = os.path.join(REPO_DIR, 'include', 'onnxruntime', 'core')
+
+        # copy necessary header files
+        shutil.copy(os.path.join(src_headers_base_path, 'session', 'onnxruntime_c_api.h'), dst_headers_path)
+        shutil.copy(os.path.join(src_headers_base_path, 'session', 'onnxruntime_cxx_api.h'), dst_headers_path)
+        shutil.copy(os.path.join(src_headers_base_path, 'session', 'onnxruntime_cxx_inline.h'), dst_headers_path)
+        shutil.copy(os.path.join(src_headers_base_path, 'providers', 'cpu', 'cpu_provider_factory.h'),
+                    dst_headers_path)
+        shutil.copy(os.path.join(src_headers_base_path, 'providers', 'coreml', 'coreml_provider_factory.h'),
```

Comment:
This is for Android isn't it, so should be the NNAPI EP factory?

Reply:
Moved the header selection to CMake to be based on the actual build settings, not manual select here